connexion/apis/abstract.py

```diff
@@ -281,6 +306,20 @@ def load_spec_from_file(self, arguments, specification):
             swagger_string = jinja2.Template(swagger_template).render(**arguments)
             return yaml.safe_load(swagger_string)  # type: dict
 
+    def _get_spec_version(self, spec):
+        try:
+            version_string = spec.get('openapi') or spec.get('swagger')
+        except AttributeError:
+            raise InvalidSpecification('Unable to get spec version')
```

Comment:
I really disagree with this. If we get this exception, then it means we couldn't find either of `openapi` / `swagger` keys in the spec. Printing the whole spec here may dump several pages of text to their terminal, and it won't be useful because it won't show them the _absence of the required field_.
I'll happily make this error more verbose though - let me know if that would address your concern.